### PR TITLE
Modernize sparsification info type hints

### PR DESCRIPTION
**Pull request by prdeepakbabu**
## Summary

- Modernize type annotations in `pytorch/utils/sparsification_info` for Python 3.10+
- Replace legacy `typing` aliases like `Dict`, `Tuple`, `Union`, `Optional`, and `List`
- Keep the change type-only with no runtime behavior changes

Part of https://github.com/vllm-project/llm-compressor/issues/1927

## Tests

- `PATH="$PWD/.venv/bin:$PATH" make quality`
- `python -m py_compile src/llmcompressor/pytorch/utils/sparsification_info/helpers.py src/llmcompressor/pytorch/utils/sparsification_info/configs.py src/llmcompressor/pytorch/utils/sparsification_info/module_sparsification_info.py`

### CODE REVIEW

**Review by kylesayrs — APPROVED** [review]

**Comment by prdeepakbabu** [comment]
I have addressed the current typing review feedback and reran local validation:

- `PATH="$PWD/.venv/bin:$PATH" make quality`
- `python -m py_compile src/llmcompressor/pytorch/utils/sparsification_info/helpers.py src/llmcompressor/pytorch/utils/sparsification_info/configs.py src/llmcompressor/pytorch/utils/sparsification_info/module_sparsification_info.py`

I tried to add the `ready` label so the full test suite can run, but I do not have permission to add labels on this repository. Could a maintainer please add `ready` when appropriate?

File: src/llmcompressor/pytorch/utils/sparsification_info/helpers.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from typing import List, Optional, Union
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (HDCharles) [reply src/llmcompressor/pytorch/utils/sparsification_info/helpers.py]: True, can be removed

File: src/llmcompressor/pytorch/utils/sparsification_info/configs.py [hunk 2]
```diff
@@ -344,7 +344,7 @@ def loggable_items(
         self,
         enabled_only: bool = False,
         **kwargs,
-    ) -> Generator[Tuple[str, Union[Dict[str, int], float, int]], None, None]:
+    ) -> Generator[tuple[str, dict[str, int] | float | int], None, None]:
```
[the remark this answers is not in the document]
Comment (HDCharles) [reply src/llmcompressor/pytorch/utils/sparsification_info/configs.py]: True

File: src/llmcompressor/pytorch/utils/sparsification_info/helpers.py [hunk 3]
```diff
@@ -9,9 +9,9 @@
 
 def get_leaf_operations(
     model: torch.nn.Module,
-    operations_to_skip: Optional[List[torch.nn.Module]] = None,
-    operations_to_unwrap: Optional[List[torch.nn.Module]] = None,
-) -> List[torch.nn.Module]:
+    operations_to_skip: list[torch.nn.Module] | None = None,
+    operations_to_unwrap: list[torch.nn.Module] | None = None,
+) -> list[torch.nn.Module]:
```
[the remark this answers is not in the document]
Comment (HDCharles) [reply src/llmcompressor/pytorch/utils/sparsification_info/helpers.py:12]: Yeah this is good to fix with the suggestion

**Review by HDCharles — CHANGES_REQUESTED** [review]

**Comment by prdeepakbabu** [comment]
Addressing the requested follow-up in `2515e215`:

- removed `from __future__ import annotations` from `helpers.py`
- made `get_leaf_operations` consistently return `list[torch.nn.Module]`
- removed the `try/except TypeError` workaround in the recursive call

Reran local validation:

- `PATH="$PWD/.venv/bin:$PATH" make quality`
- `python -m py_compile src/llmcompressor/pytorch/utils/sparsification_info/helpers.py src/llmcompressor/pytorch/utils/sparsification_info/configs.py src/llmcompressor/pytorch/utils/sparsification_info/module_sparsification_info.py`

**Review by HDCharles — APPROVED** [review]
